plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -331,14 +338,76 @@ fu_thunderbolt_device_setup (FuDevice *device, GError **error)
 		fu_device_add_flag (device, FWUPD_DEVICE_FLAG_USABLE_DURING_UPDATE);
 		/* forces the device to write to authenticate on disconnect attribute */
 		fu_device_remove_flag (device, FWUPD_DEVICE_FLAG_SKIPS_RESTART);
-	} else {
-		self->auth_method = "nvm_authenticate";
 	}
 
-	/* success */
 	return TRUE;
 }
 
+static gboolean
+fu_thunderbolt_device_setup_retimer (FuDevice *device, GError **error)
+{
+	FuThunderboltDevice *self = FU_THUNDERBOLT_DEVICE (device);
+	guint16 did;
+	guint16 vid;
+	g_autofree gchar *idx = g_path_get_basename (self->devpath);
+	g_autofree gchar *instance = NULL;
+
+	fu_device_set_physical_id (device, idx);
+	fu_device_set_name (device, "Retimer");
+	fu_device_add_flag (device, FWUPD_DEVICE_FLAG_UPDATABLE);
+	fu_device_add_flag (device, FWUPD_DEVICE_FLAG_DUAL_IMAGE);
+
+	vid = fu_udev_device_get_vendor (FU_UDEV_DEVICE (self));
+	if (vid == 0x0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "missing vendor id");
+		return FALSE;
+	}
+
+	did = fu_udev_device_get_model (FU_UDEV_DEVICE (self));
+	if (did == 0x0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "missing device id");
+		return FALSE;
+
+	}
+
+	instance = g_strdup_printf ("TBT-%04x%04x-retimer%s",
+				    (guint) vid,
+				    (guint) did,
+				    idx);
+	fu_device_add_instance_id (device, instance);
+
+	/* hardcoded for now - we don't recognize anyone else yet */
+	fu_device_set_vendor (device, "Intel");
```

Comment:
I think this should already be set from the `ID_VENDOR_FROM_DATABASE` key in `FuUdevDevice->probe`, no?

Reply:
I don't know, I've never seen that key set in the `thunderbolt` subsystem.  I've adjusted to only set if it's not set.